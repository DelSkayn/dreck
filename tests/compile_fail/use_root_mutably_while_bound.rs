use dreck::*;

fn use_root<'own>(owner: &Owner<'own>, root: &mut Root<'own>, _: Gc<'_, 'own, i32>) {
    root.collect_full(owner);
}

fn main() {
    new_root!(owner, root);

    let v = root.add(0);
    use_root(&owner, root, v);
}
