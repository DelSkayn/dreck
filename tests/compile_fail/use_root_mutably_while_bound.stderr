error[E0502]: cannot borrow `*root` as mutable because it is also borrowed as immutable
  --> tests/compile_fail/use_root_mutably_while_bound.rs:11:5
   |
10 |     let v = root.add(0);
   |             ----------- immutable borrow occurs here
11 |     use_root(&owner, root, v);
   |     --------^^^^^^^^^^^^^^^^^
   |     |
   |     mutable borrow occurs here
   |     immutable borrow later used by call
